rle/rleplus_writer.go

```diff
@@ -2,14 +2,20 @@ package rlepluslazy
 
 import (
 	"encoding/binary"
+	"errors"
 )
 
+var ErrSameValRuns = errors.New("2 consecutive runs with the same value")
+
 func EncodeRuns(rit RunIterator, buf []byte) ([]byte, error) {
+	rit = newNormIter(rit)
```

Comment:
Q: Do we ever care about trailing 0 runs when encoding?

Reply:
No, we don't care about trailing zeros.